mlflow/projects/kubernetes.py

```diff
@@ -0,0 +1,102 @@
+from __future__ import absolute_import
+import yaml
+import logging
+import docker
+import os
+import time
+import kubernetes
+from datetime import datetime
+from mlflow.exceptions import ExecutionException
+
+_logger = logging.getLogger(__name__)
+
+
+def push_image_to_registry(image_uri):
+    client = docker.from_env()
+    _logger.info("=== Pushing docker image %s ===", image_uri)
+    for line in client.images.push(repository=image_uri, stream=True, decode=True):
+        if 'error' in line and line['error']:
+            raise ExecutionException("Error while pushing to docker registry: "
+                                     "{error}".format(error=line['error']))
+
+
+def _get_kubernetes_job_definition(image, command, env_vars, job_template):
+    timestamp = datetime.now().strftime('%Y-%m-%d-%H-%M-%S-%f')
+    job_name = "{}-{}".format(image.split('/')[-1].replace(':', '-'), timestamp)
+    _logger.info("=== Creating Job %s ===", job_name)
+    enviroment_variables = ""
+    if os.environ.get('AZURE_STORAGE_ACCESS_KEY'):
```

Comment:
Could we provide these through the kube-job-template for now?

Reply:
Sure, I removed it.

We can let users provide it in base image so they can build it with these variables values. Thinking about security this is a better choice because these keys wouldn't be exposed in config files. What do you think?